python-spec/src/somacore/collection.py

```diff
@@ -135,22 +213,30 @@ def uri(self) -> str:
     def metadata(self) -> Dict[str, Any]:
         return self._metadata
 
-    def add(
-        self,
-        key: str,
-        cls: Type[_ST],
-        *,
-        uri: Optional[str] = None,
-        platform_config: Optional[options.PlatformConfig] = None,
-    ) -> _ST:
-        del key, cls, uri, platform_config  # All unused
+    @classmethod
+    def open(cls, *args, **kwargs) -> NoReturn:
+        del args, kwargs  # All unused
+        raise TypeError("SimpleCollections are in-memory only and cannot be opened.")
+
+    @classmethod
+    def create(cls, *args, **kwargs) -> "SimpleCollection":
+        del args, kwargs  # All unused
+        # SimpleCollection is in-memory only, so just return a new empty one.
+        return cls()
+
+    def add_collection(self, *args, **kwargs) -> NoReturn:
+        del args, kwargs  # All unused
         # TODO: Should we be willing to create Collection-based child elements,
```

Comment:
I'm struggling to answer this question, as I'm unsure what the use case is for SimpleCollection.  So I think, for now, your simplest-possible-implementation is a good one.